src/odemis/acq/calibration.py

```diff
@@ -356,48 +357,59 @@ def apply_spectrum_corrections(data, bckg=None, coef=None):
+def write_trigger_delay_csv(filename, trig_delays):
     """
-    Reads the time range and trigger delay values from a csv object.
-    Checks values for validity.
-    :parameter data: (csv.reader object) calibration file
-    :parameter timeRange_choices: (frozenset) choices possible for timeRange VA
-    :parameter triggerDelay_range: (tuple) range possible for trigger delay values
-    :return: (dict) new dictionary containing the loaded time range to trigger delay info
+    Store the MD_TIME_RANGE_TO_DELAY into a CSV file
+    filename (str): the path to file (if it already exists, it will be overwritten)
+    trig_delays (dict float -> float): time range to trigger delay info
     """
-    new_dict = {}
-
-    for timeRange, delay in data:
-
-        try:
-            timeRange = float(timeRange)
-            delay = float(delay)
-        except ValueError:
-            raise ValueError("Trigger delay %s and/or time range %s is not of type float. "
-                             "Please check calibration file for trigger delay." % (delay, timeRange))
-
-        # check delay in range allowed
-        if not triggerDelay_range[0] <= delay <= triggerDelay_range[1]:
-            raise ValueError("Trigger delay %s corresponding to time range %s is not in range %s. "
-                             "Please check the calibration file for the trigger delay." %
-                             (delay, timeRange, triggerDelay_range))
-
-        # check timeRange is in possible choices for timeRange on HW
-        choice = find_closest(timeRange, timeRange_choices)
-        if not almost_equal(timeRange, choice):
-            raise ValueError("Time range % s found in calibration file is not a possible choice "
-                             "for the time range of the streak unit. "
-                             "Please modify csv file so it fits the possible choices for the "
-                             "time range of the streak unit. "
-                             "Values in file must be of format timeRange:triggerDelay (per line)."
-                             % timeRange)
-
-        new_dict[timeRange] = delay
+
+    with open(filename, 'w') as csvfile:
+        calibFile = csv.writer(csvfile, delimiter=':')
+        for time_range, trig_delay in trig_delays.items():
+            calibFile.writerow([time_range, trig_delay])
+
+
+def read_trigger_delay_csv(filename, time_choices, trigger_delay_range):
+    """
+    Read the MD_TIME_RANGE_TO_DELAY from a CSV file, and check its validity based on the hardware
+    filename (str): the path to file
+    trig_delays (dict float -> float): time range to trigger delay info
+    time_choices (set): choices possible for timeRange VA
+    return (dict float -> float): new dictionary containing the loaded time range to trigger delay info
+    raise ValueError: if the data of the CSV file cannot be parsed or doesn't fit the hardware
```

Comment:
What happens if the file doesn't exist?

Reply:
IOError. I've added it to the docstring.